edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/ExperimentalFeatures.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Service
+{
+    using Microsoft.Azure.Devices.Edge.Storage;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.Logging;
+
+    public class ExperimentalFeatures
+    {
+        ExperimentalFeatures(bool enabled, bool disableCloudSubscriptions, bool enableUploadLogs, bool enableGetLogs)
+        {
+            this.Enabled = enabled;
+            this.DisableCloudSubscriptions = disableCloudSubscriptions;
+            this.EnableUploadLogs = enableUploadLogs;
+            this.EnableGetLogs = enableGetLogs;
+        }
+
+        public static ExperimentalFeatures Init(IConfiguration experimentalFeaturesConfig, ILogger logger)
+        {
+            bool enabled = experimentalFeaturesConfig.GetValue("enabled", false);
+            bool disableCloudSubscriptions = enabled && experimentalFeaturesConfig.GetValue("disableCloudSubscriptions", false);
+            bool enableUploadLogs = enabled && experimentalFeaturesConfig.GetValue("enableUploadLogs", false);
+            bool enableGetLogs = enabled && experimentalFeaturesConfig.GetValue("enableGetLogs", false);
+            var experimentalFeatures = new ExperimentalFeatures(enabled, disableCloudSubscriptions, enableUploadLogs, enableGetLogs);
+            logger.LogInformation($"Experimental features configuration: {experimentalFeatures.ToJson()}");
+            return experimentalFeatures;
+        }
+
+        public bool Enabled { get; }
```

Comment:
>Enabled [](start = 20, length = 7)

nit: IsEnabled, same for other features

Reply:
I was going for consistency here - most of the other enabled flags are just named Enabled (for example - MqttSettings__Enabled, AmqpSetting__Enabled).

Reply:
What I meant was IsEnabled for the property name, not for the config flag